### PR TITLE
test: comprehensive test suite review — ACE naming, quality uplift, dedup

### PR DESCRIPTION
**Pull request by seanb4t**
## Summary

Comprehensive review of ~210 unit test files across 37 packages:

- **Naming:** Applied ACE (Action, Condition, Expectation) sentence-style naming convention to ~600+ test functions. Test names now read as behavior specifications when processed by `gotestdox`.
- **Quality:** Fixed ~15 zero-assertion "don't panic" tests, added ~20 new tests for uncovered error paths, expanded validation coverage.
- **Deduplication:** Created shared `assertProviderContract` helper for 14 attribute provider test files, eliminating ~83 lines of boilerplate. Removed experimental `ast_spike_test.go` (migrated 2 unique tests to `parser_test.go`). Removed 1 duplicate migration test.
- **Standards:** Added Test Naming and Test Quality sections to CLAUDE.md for all future development.

### Packages covered (all phases)

| Phase | Packages | Key changes |
|-------|----------|-------------|
| 1 | xdg, tls, naming, observability, idgen, logging, telemetry | Convention established |
| 2 | core, store, config, control | Large file renames (migrate, grpc_server) |
| 3 | auth | Guest service error path tests added |
| 4 | command, handlers | Zero-assertion metrics test fixed, validation expanded |
| 5 | plugin, hostfunc, lua, goplugin | 157 renames across 10 files |
| 6 | world, world/postgres | Orphan detector error paths tested |
| 7 | access, policy, attribute, audit, dsl | Contract test helper, spike removal |
| misc | bootstrap, property, content, session, telnet, web, grpc, cmd, pkg | Bootstrap dedup audit (no overlap found) |

### Commit structure

17 commits, alternating rename-only (mechanical, easy review) and quality (real review) per phase. All will be squash-merged.

## Test plan

- [x] `task pr-prep` passes (all CI gates)
- [x] 5,052 unit tests pass
- [x] 277 integration tests pass
- [x] 48 E2E tests pass
- [x] 0 lint issues
- [ ] Verify `gotestdox` output reads as sentences for sample packages

**Spec:** `docs/superpowers/specs/2026-04-04-test-suite-review-design.md`
**Plan:** `docs/superpowers/plans/2026-04-04-test-suite-review.md`

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by CodeRabbit

* **Documentation**
  * Added formal Go test conventions and quality standards (ACE naming, assertion and coverage guidance).
  * Added a phased test-suite review plan and design to standardize ~210 tests.
  * Added server lifecycle & health documentation covering startup phases, readiness, and health tiers.

* **Tests**
  * Standardized test naming and organization across the codebase.
  * Consolidated provider contract checks and added helpers for consistent validations.
  * Added and improved tests for edge cases, error-code assertions, and several service/component behaviors.

### CODE REVIEW

File: site/docs/contributing/lifecycle-and-health.md [hunk 1]
```diff
@@ -0,0 +1,261 @@
...
+
+**Safety net:** A background poller queries the database every 10 seconds for
+changes. This catches anything the fast path missed — manual database edits,
+multi-instance deployments, or edge cases where the in-process invalidation
+didn't fire.
+
+The health tracker watches both paths. Successful reloads keep the tier at
+Warm. Failures escalate through the tiers. When the tier reaches Stale, the
+tracker tells the ABAC engine to enter degraded mode — all non-system requests
+are denied until recovery.
```
[the remark this answers is not in the document]
Comment (seanb4t) [reply site/docs/contributing/lifecycle-and-health.md:241]: mistakenly included in this pr, removing

**Comment by seanb4t** [comment]
## CodeRabbit Fixes Applied

Fixed 4 files based on 5 unresolved review comments.

**Files modified:**
- `CLAUDE.md` — lowercase subtest names in example
- `internal/access/policy/attribute/exit_test.go` — exit-specific contract assertions
- `internal/plugin/lua/state_internal_test.go` — stale comment fix
- `internal/world/property_lifecycle_test.go` — rename misleading test + add error code assertion

**Commit:** `6ab0c1a`
